#! /bin/sh -e 
## <PATCHNAME>.dpatch by <PATCH_AUTHOR@EMAI>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Description: coverity: tty_ldisc_ref return null check
## DP: Patch author: KAMBAROV, ZAUR <kambarov@berkeley.edu>
## DP: Upstream status: upstream

. $(dirname $0)/DPATCH

@DPATCH@
From: KAMBAROV, ZAUR <kambarov@berkeley.edu>
Date: Thu, 30 Jun 2005 01:12:05 +0000 (-0700)
Subject: [PATCH] coverity: tty_ldisc_ref return null check
X-Git-Tag: v2.6.12.3
X-Git-Url: http://www.kernel.org/git/?p=linux/kernel/git/gregkh/linux-2.6.12.y.git;a=commitdiff;h=19a51b0ee71ee00497619f441c59b4acd44cf1e5

  [PATCH] coverity: tty_ldisc_ref return null check
  
  We add a check of the return value of tty_ldisc_ref(), which
  is checked 7 out of 8 times, e.g.:
  
  149  		ld = tty_ldisc_ref(tty);
  150  		if (ld != NULL) {
  151  			if (ld->set_termios)
  152  				(ld->set_termios)(tty, &old_termios);
  153  			tty_ldisc_deref(ld);
  154  		}
  
  This defect was found automatically by Coverity Prevent, a static analysis
  tool.
  
  (akpm: presumably `ld' is never NULL.  Oh well)
  
  Signed-off-by: Zaur Kambarov <zkambarov@coverity.com>
  Acked-by: Alan Cox <alan@lxorguk.ukuu.org.uk>
  Signed-off-by: Andrew Morton <akpm@osdl.org>
  Signed-off-by: Chris Wright <chrisw@osdl.org>
  Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

Backported to Debian's kernel-source-2.6.8 by dann frazier <dannf@debian.org>

--- a/drivers/char/tty_ioctl.c
+++ b/drivers/char/tty_ioctl.c
@@ -476,11 +476,11 @@ int n_tty_ioctl(struct tty_struct * tty,
 			ld = tty_ldisc_ref(tty);
 			switch (arg) {
 			case TCIFLUSH:
-				if (ld->flush_buffer)
+				if (ld && ld->flush_buffer)
 					ld->flush_buffer(tty);
 				break;
 			case TCIOFLUSH:
-				if (ld->flush_buffer)
+				if (ld && ld->flush_buffer)
 					ld->flush_buffer(tty);
 				/* fall through */
 			case TCOFLUSH:
